dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -74,12 +76,50 @@ public CommunityRest findOne(Context context, UUID id) {
 		Page<CommunityRest> page = new PageImpl<Community>(communities, pageable, total).map(converter);
 		return page;
 	}
+
+	// TODO: Add methods in dspace api to support pagination of top level
+	// communities
+	@SearchRestMethod(name="top")
+	public Page<CommunityRest> findAllTop(Pageable pageable) {
+		List<Community> topCommunities = new ArrayList<Community>();
+		int total = 0;
+		try {
+			List<Community> it = cs.findAllTop(obtainContext());
+			total = it.size();
+			for (Community c : it) {
+				topCommunities.add(c);
```

Comment:
Can't you use `topCommunities.addAll(it)` here?

Reply:
indeed it can be simplified. We don't need two lists at all